plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/TrinoCatalogFactory.java

```diff
@@ -75,7 +76,7 @@ public TrinoCatalog create()
     {
         switch (catalogType) {
             case HIVE:
-                return new TrinoHiveCatalog(catalogName, metastore, hdfsEnvironment, typeManager, tableOperationsProvider, trinoVersion, isUniqueTableLocation);
+                return new TrinoHiveCatalog(catalogName, memoizeMetastore(metastore, 1000), hdfsEnvironment, typeManager, tableOperationsProvider, trinoVersion, isUniqueTableLocation);
```

Comment:
Can we have a config like we have for Hive connector ?

Reply:
i discussed this with @losipiuk and we determined it's OK as is.